Rakefile

```diff
@@ -152,3 +160,36 @@ CLEAN.include(BUILDDIR)
 CLEAN.include(PKGDIR)
 
 task :default => [:rubocop, :spec, 'pkg:generate_source']
+
+if pin_task
+  desc 'Pin all the modules in Puppetfile to released versions instead of git branches'
+  task :pin_modules do
+    filename = 'Puppetfile'
+    $new_content = []
```

Comment:
Why a global variable?

Reply:
Basically I load `Puppetfile` as a ruby script and handle `forge` and `mod` as ruby functions. Because I want to buffer the output somewhere and can't modify the Puppetfile to store it somewhere, this is the only way I knew how to do this. If there's a better way, I'd love to hear it.

Reply:
Oh I see - clever.